ENG-879: Implement pipeline for releasing dockers

Sudokamikaze opened the pull request:
Test pipeline job available at rel.cd

EvgeniyPatlan commented on docker/percona-docker.groovy:
```diff
@@ -0,0 +1,131 @@
+            steps {
+                sh '''
+                    sudo bash -x docker_builder.sh --builddir=$(pwd)/tmp --export_container=1 --product=${DOCKER_NAME} --organization=${DOCKER_ORG} --version=${DOCKER_VERSION}
+                    sudo chmod -R 777 ${WORKSPACE}/tarball
```
why do we need such permissions?

Sudokamikaze replied:
As docker processes running as root, jenkins default user can't access/modify/read files without such permission

EvgeniyPatlan replied:
you can do 
```
sudo usermod -aG docker jenkins
sh 'sg docker -c "...." '
```

Sudokamikaze replied:
Done

Sudokamikaze commented:
> also please provide the link to the job

https://rel.cd.percona.com/job/Dockers-testingground/

Sudokamikaze commented:
In latest force push:
* Added gzip support for archiving images
* Reworked permissions
* Cleaned codebase, removed unneeded flags and reworked few as choice parameters instead of string
* Implemented tests
